src/dbnode/storage/namespace.go

```diff
@@ -393,6 +397,18 @@ func (n *dbNamespace) Shards() []Shard {
 	return databaseShards
 }
 
+func (n *dbNamespace) SchemaRegistry() namespace.SchemaRegistry {
+	n.RLock()
+	defer n.RUnlock()
+	return n.schemaRegistry
+}
+
+func (n *dbNamespace) SetSchemaRegistry(v namespace.SchemaRegistry) {
+	n.Lock()
+	defer n.Unlock()
+	n.schemaRegistry = v
```

Comment:
Hm should we not do some validation here?

Reply:
What happens if the validation fails during loading? Will the new namespaces still propagate to the clustered database or will it error out before then?

Reply:
It will not be propagated (we won't update schema registry on the namespace). Ideally, we catch validation error before it get persisted into etcd.

Reply:
👍